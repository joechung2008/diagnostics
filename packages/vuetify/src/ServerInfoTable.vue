<script lang="ts" setup>
import type { ServerInfoProps } from './ServerInfoTable.types'

const { hostname, uptime, serverId, deploymentId, nodeVersions, extensionSync } =
  defineProps<ServerInfoProps>()
</script>

<template>
  <v-table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Value</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Hostname</td>
        <td>{{ hostname }}</td>
      </tr>
      <tr>
        <td>Uptime</td>
        <td>{{ uptime }}</td>
      </tr>
      <tr>
        <td>Server ID</td>
        <td>{{ serverId }}</td>
      </tr>
      <tr>
        <td>Deployment ID</td>
        <td>{{ deploymentId }}</td>
      </tr>
      <tr>
        <td>Node Versions</td>
        <td>{{ nodeVersions }}</td>
      </tr>
      <tr>
        <td>Extension Sync | Total Sync All Count</td>
        <td>{{ extensionSync.totalSyncAllCount }}</td>
      </tr>
    </tbody>
  </v-table>
</template>
