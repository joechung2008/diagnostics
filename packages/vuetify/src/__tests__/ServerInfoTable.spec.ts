import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import ServerInfoTable from '../ServerInfoTable.vue'

describe('ServerInfoTable', () => {
  it('renders server info', () => {
    const wrapper = mount(ServerInfoTable, {
      props: {
        hostname: 'my-host',
        uptime: 24,
        serverId: 'srv-123',
        deploymentId: 'dep-456',
        nodeVersions: 'v18.0.0',
        extensionSync: { totalSyncAllCount: 42 },
      },
    })
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
