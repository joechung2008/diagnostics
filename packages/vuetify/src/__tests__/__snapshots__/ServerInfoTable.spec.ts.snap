// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfoTable > renders server info 1`] = `
"<div class="v-table v-theme--light v-table--density-default">
  <!---->
  <div class="v-table__wrapper">
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Hostname</td>
          <td>my-host</td>
        </tr>
        <tr>
          <td>Uptime</td>
          <td>24</td>
        </tr>
        <tr>
          <td>Server ID</td>
          <td>srv-123</td>
        </tr>
        <tr>
          <td>Deployment ID</td>
          <td>dep-456</td>
        </tr>
        <tr>
          <td>Node Versions</td>
          <td>v18.0.0</td>
        </tr>
        <tr>
          <td>Extension Sync | Total Sync All Count</td>
          <td>42</td>
        </tr>
      </tbody>
    </table>
  </div>
  <!---->
</div>"
`;
